### PR TITLE
m

### PR DESCRIPTION
**Pull request by shortcut1337**

### CODE REVIEW

File: __manifest__.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 {
     'name': "Approvals Statement",
     'version': '1.3',
-    'depends': ['account_accountant'],
+    'depends': ['account_accountant', 'sale_stock'],
```
Comment (shortcut1337) [inline __manifest__.py:6]: @sbuhl un soucis avec ce depends? la db a le module de toute facon?

File: models/account_approval.py [hunk 2]
```diff
@@ -1,9 +1,51 @@
...
+
+
+class DiscountSettings(models.Model):
+    _name = 'discount.settings'
+    _description = 'Discount Settings'
+
+    name = fields.Char()
+    max_amount = fields.Integer(required=True)
+    user_ids = fields.One2many('res.users', 'discount_settings_id')
+    # TODO: Maybe you want a group_ids here to allow a whole group in once?
```
Comment (shortcut1337) [inline models/account_approval.py]: @sbuhl

File: models/account_approval.py [hunk 3]
```diff
@@ -1,9 +1,51 @@
...
+
+    discount_settings_id = fields.Many2one('discount.settings')
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    @api.constrains('discount')
+    def _check_max_allowed_discount(self):
+        # TODO: Maybe we can authorize 999% here if group admin? instead of
```
Comment (shortcut1337) [inline models/account_approval.py]: @sbuhl

**Review by shortcut1337 — COMMENTED** [review]
Working version, à tester et répondre aux 2 points pour voir de là le besoin exacte

File: __manifest__.py (hunk 1 above)
Comment (sbuhl) [reply __manifest__.py:6]: @shortcut1337 No objection to use it.

File: models/account_approval.py (hunk 2 above)
Comment (sbuhl) [reply models/account_approval.py]: @shortcut1337 Maybe more straightforward to allow a whole group in once, yes.

File: models/account_approval.py (hunk 3 above)
Comment (sbuhl) [reply models/account_approval.py]: @shortcut1337 Ok

File: models/account_approval.py (hunk 2 above)
Comment (sbuhl) [reply models/account_approval.py]: As said in another comment,
Let's use the groups, that I still have to create: 
1. Junior
2. Mid-Level
3. Senior
4. Store Manager
5. Director

Comment (sbuhl) [reply models/account_approval.py]: Here the groups that I created:
[security/security.xml](https://github.com/sbuhl/account_approval/commit/8539a4b4eb6e33dc4daf00a6c480f51b103d88e0)

File: data/discount_settings.xml [hunk 4]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0"?>
+<odoo>
+	<data noupdate="1">
+        <record id="discount_settings_debutant" model="discount.settings">
```
Comment (sbuhl) [inline data/discount_settings.xml]: Can you change and use the same name as the groups created here:
https://github.com/sbuhl/account_approval/commit/8539a4b4eb6e33dc4daf00a6c480f51b103d88e0

Junior: 0
Mid-Level: 0
Advanced: 5
Store Manager: 20
Director: 999

Comment (shortcut1337) [reply data/discount_settings.xml]: Adapted

**Pull request merged**